File: src/main/java/pl/training/workshop/m6/s02_polymorphism/step1/MarathonScreening.java
package pl.training.workshop.m6.s02_polymorphism.step1;

import pl.training.workshop.shared.Money;

/** Krok 1: pierwsza podklasa - maraton ma własne, nazwane dane (films zamiast value). */
public final class MarathonScreening extends Screening {
    private final int films;

    MarathonScreening(String title, int films) {
        super(Kind.MARATHON, title, films);
        this.films = films;
    }

    @Override
    public String label() {
        return "Maraton: " + title() + " (" + films + " filmy)";
    }

    @Override
    public int durationMinutes() {
        return films * 120 + (films - 1) * 15;
    }

    @Override
    public Money price() {
        return Money.of("20.00").times(films);
    }
}

File: csharp/src/Training.Workshop/M6/S02Polymorphism/Step1/MarathonScreening.cs
using Training.Workshop.Shared;

namespace Training.Workshop.M6.S02Polymorphism.Step1;

/// <summary>Krok 1: pierwsza podklasa - maraton ma własne, nazwane dane (_films zamiast _value).</summary>
public sealed class MarathonScreening : Screening
{
    private readonly int _films;

    internal MarathonScreening(string title, int films)
        : base(Kind.Marathon, title, films)
    {
        _films = films;
    }

    public override string Label()
    {
        return "Maraton: " + Title + " (" + _films + " filmy)";
    }

    public override int DurationMinutes()
    {
        return _films * 120 + (_films - 1) * 15;
    }

    public override Money Price()
    {
        return Money.Of("20.00").Times(_films);
    }
}
